lib/cocoapods/target/aggregate_target.rb

```diff
@@ -23,6 +26,21 @@ def initialize(target_definition, sandbox)
       @xcconfigs = {}
     end
 
+    # @return True if the user_target's pods are
+    # for an extension and must be embedded in a host,
+    # target, otherwise false.
+    #
+    def requires_host_target?
+      # If we don't have a user_project, then we can't
+      # glean any info about how this target is going to
+      # be integrated, so return false since we can't know
+      # for sure that this target refers to an extension
+      # target that would require a host target
+      return false if user_project.nil?
+      raise ArgumentError, "Expected single user_target for #{name}" unless user_targets.count == 1
```

Comment:
You mean instead of validating `user_targets.count`, check `user_targets.map(&:symbol_type).uniq.count`?